File: Daily task/2069-M-Walking-Robot-Simulation-II/walking_robot_simu_ii.py
class Robot:
    def __init__(self, width: int, height: int):
        self.width = width
        self.height = height
        self.flag=0
        self.position = 0

    def step(self, num: int) -> None:
        loop_len = (self.width + self.height) * 2 - 4
        self.position = (self.position + num) % loop_len
        flag = 1

    def getPos(self) -> list[int]:
        result = [0, 0]
        if self.position >= 0 and self.position <= self.width - 1:
            result[0] = self.position
            result[1] = 0
        elif self.position >= self.width and self.position < self.width + self.height - 1:
            result[0] = self.width-1
            result[1] = self.position - (self.width - 1)
        elif self.position >= self.width + self.height - 1 and self.position < 2 * self.width + self.height - 2:
            result[0] = (self.width - 1) - (self.position - (self.width + self.height - 2))
            result[1] = self.height - 1
        else:
            result[0] = 0
            result[1] = (self.height - 1) - (self.position - (2 * self.width + self.height - 3))
        return result

    def getDir(self) -> str:
        if self.flag==1 and self.position == 0:
            return "South"
        
        if self.position >= 0 and self.position <= self.width - 1:
            return "East"
        elif self.position >= self.width and self.position < self.width + self.height - 1:
            return "North"
        elif self.position >= self.width + self.height - 1 and self.position < 2 * self.width + self.height - 2:
            return "West"
        else:
            return "South"

# Your Robot object will be instantiated and called as such:
# obj = Robot(width, height)
# obj.step(num)
# param_2 = obj.getPos()
# param_3 = obj.getDir()

File: Daily task/2069-M-Walking-Robot-Simulation-II/walking_robot_simu_ii.cpp
class Robot {
    int width;
    int height;
    int position;
    int x;
    int y;
    int flag;
    string direction;

    public:
        Robot(int width, int height, int position=0, int x=0, int y=0, string direction="East", int flag=0) {
            this->width = width;
            this->height = height;
            this->position = position;
            this->x = x;
            this->y=y;
            this->direction=direction;
            this->flag=flag;
        }
        
        void step(int num) {
            int loop_len = (width + height) * 2 - 4;
            position = (num + position) % loop_len;
            flag = 1;
        }
        
        vector<int> getPos() {
            vector<int> result(2, 0);

            if (0<= position && position <= width-1) {
                result[0] = position;
                result[1] = 0;
                // direction = "East";
            } else if (width <= position && position < width + height - 1) {
                result[0] = width - 1;
                result[1] = position - (width-1);
                // direction = "North";

            } else if (width+height-1 <= position && position < 2*width+height-2) {
                result[0] = (width-1) - (position-(width+height-2));
                result[1] = height-1;
                // direction = "West";
            } else if (2*width+height-2 <= position && position < (width+height)*2) {
                result[0] = 0;
                result[1] = (height-1)-(position-(2*width+height-3));
                // direction = "South";
            }
            return result;
        }
        
        string getDir() {
            if (0<= position && position <= width-1) {
                direction = "East";
            } else if (width <= position && position < width + height - 1) {
                direction = "North";

            } else if (width+height-1 <= position && position < 2*width+height-2) {
                direction = "West";
            } else if (2*width+height-2 <= position && position < (width+height)*2) {
                direction = "South";
            }

            if (flag && position==0)
                return "South";
            return direction;
        }
};

/**
 * Your Robot object will be instantiated and called as such:
 * Robot* obj = new Robot(width, height);
 * obj->step(num);
 * vector<int> param_2 = obj->getPos();
 * string param_3 = obj->getDir();
 */